environment/src/main/java/com/sequenceiq/environment/environment/service/FreeipaService.java

```diff
@@ -0,0 +1,55 @@
+package com.sequenceiq.environment.environment.service;
+
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import com.dyngr.Polling;
+import com.dyngr.core.AttemptMaker;
+import com.sequenceiq.freeipa.api.v1.freeipa.stack.FreeIpaV1Endpoint;
+import com.sequenceiq.freeipa.api.v1.freeipa.stack.model.describe.DescribeFreeIpaResponse;
+
+@Service
+public class FreeipaService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(FreeipaService.class);
+
+    @Value("${env.stop.polling.attempt:90}")
+    private Integer attempt;
+
+    @Value("${env.stop.polling.sleep.time:5}")
```

Comment:
5s is too frequent IMHO

Reply:
FreeIpa stop finishes quickly generally. I think it is reasonable.